include/programs.h

```diff
@@ -58,7 +65,8 @@ class CommandLine {
 	std::list<std::string> cmds = {};
 	std::string file_name = "";
 
-	bool FindEntry(char const * const name,cmd_it & it,bool neednext=false);
+	bool FindEntry(char const * const name,cmd_it &it, bool neednext = false);
```

Comment:
Is it possible to use a `const std::string & name` instead of a C-string pointer?

Reply:
Regarding the command iterator, if we're finding items in it, then this is a read-only operation (and not be allowed to change the contents).

So we could use this (adding it to `programs.h`):

``` c++
using const_cmd_it = std::list<std::string>::const_iterator;
```

And then take it in as a value:

 ``` c++
bool FindEntry(onst std::string & name, const_cmd_it it, bool need_next = false);
```

Reply:
Sure! The PR now actually tries to cleanup this function too.